### PR TITLE
New extension: Show picture size

### PR DESCRIPTION
**Pull request by TiMESPLiNTER**
Added a new extension which displays the picture size of pictures in photo and photo set posts in the upper right corner of the picture.

### CODE REVIEW

**Comment by Wolvan** [comment]
I don't wanna come off as rude, but what is this extension for? As in, what are usage examples

**Comment by TiMESPLiNTER** [comment]
@Wolvan it displays the resolution of pictures in media posts in the upper right corner of the picture. I use this extension for myself because I only want to reblog high-res photos. And instead of verify each picture I can see it right in the corner if the pic matches my "resolution requirements".

**Comment by Wolvan** [comment]
I'm unsure if this should be it's own extension. I'll run this past the team, if you don't mind

**Comment by TiMESPLiNTER** [comment]
No problem... just thought I would share it here :-). But if not its own extension what do you recommend otherwise?

**Comment by Wolvan** [comment]
That's what I want to discuss with the team. maybe a tweak

**Comment by hobinjk** [comment]
I think this is a valuable extension to include and that it has functionality beyond something that could be folded into Tweaks. Thank you for your excellent work!

**Comment by ThePsionic** [comment]
Yep, :+1: for own extension. It's new functionality that doesn't quite fit the Tweaks header imo.

**Comment by Wolvan** [comment]
It doesn't have to be tweaks, it was just the first thing that came to mind

File: Extensions/show_picture_size.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+//* TITLE Show picture size **//
```
Comment (hobinjk) [inline Extensions/show_picture_size.js]: Should be "Show Picture Size" (title cased) for consistency with other extensions.

**Comment by hobinjk** [comment]
@TiMESPLiNTER Sorry for the delay in reviewing. The functionality looks great! My one thing I would like fixed is the title-casing of the title, which I've also commented on inline.

File: Extensions/show_picture_size.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        XKit.tools.init_css('show_picture_size');
+
+        XKit.post_listener.add('show_picture_size', XKit.extensions.show_picture_size.post_listener);
+    },
+
+    destroy: function() {
+        $('.post.xkit-show-picture-size').removeClass('xkit-show-picture-size');
+        XKit.tools.remove_css('show_picture_size');
+        XKit.post_listener.remove('post_listener');
```
Comment (hobinjk) [inline Extensions/show_picture_size.js]: Should be `XKit.post_listener.remove('show_picture_size')`

**Comment by TiMESPLiNTER** [comment]
Okay I've fixed those two things. I use the extension now for about a week and it works great. Although sometimes I have to scroll some time until the post listener fires and the image sizes get rendered.

What's to proper way to ensure that the listener fires as soon as I refresh the page and not after some scrolling.

**Comment by Wolvan** [comment]
You have to also run the function manually after registering the post listener

**Comment by TiMESPLiNTER** [comment]
thx @Wolvan. Added the missing manual call of the listener method in the most recent commit.

**Comment by hobinjk** [comment]
Tested locally, LGTM. Thank you for the great work!
@homu r+

**Pull request merged**